Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -854,6 +866,9 @@ void RenderElement::styleWillChange(StyleDifference diff, const RenderStyle& new
             downcast<RenderBox>(*this).removeFloatingOrPositionedChildFromBlockLists();
         }
 
+        if (m_style.contentVisibility() != newStyle.contentVisibility() && isTopLayerAncestor())
+            document().renderView()->layer()->dirtyZOrderLists();
```

Comment:
Seems pretty random. Would be clearer if the invalidation had some name that related to top layer things.

Reply:
Indeed this was not nice, new patch removes this code luckily.